x/data/server/features/msg_define_resolver.feature

```diff
@@ -14,3 +14,16 @@ Feature: Msg/DefineResolver
       Given alice has defined a resolver with url "https://foo.bar"
       When alice attempts to define a resolver with url "https://foo.bar"
       Then expect the error "a resolver with the same URL and manager already exists: unique key violation"
+
+  Rule: Event is emitted
+
+    Scenario: EventDefineResolver is emitted
+      Given alice has defined a resolver with url "https://foo.bar"
+      When bob attempts to define a resolver with url "https://foo.bar"
+      Then expect the resolver with id "2" and url "https://foo.bar" and manager bob
+      And expect event with properties
+      """
+      {
+        "id": 1
+      }
+      """
```

Comment:
this seems.. wrong? but this is how it passes 🤔

Reply:
It looks like it's reading the event from here:

```
func (s *defineResolverSuite) AliceHasDefinedAResolverWithUrl(a string) {
	_, err := s.server.DefineResolver(s.ctx, &data.MsgDefineResolver{
		Manager:     s.alice.String(),
		ResolverUrl: a,
	})
	require.NoError(s.t, err)
}
```

Reply:
How we write tests have evolved since these initial tests. This step should probably be a resolver exists with manager, etc. and we would insert a resolver rather than call the define resolver server method.

Reply:
```feature
      Given a resolver with properties
     """
     ...
     """
      When bob attempts to define a resolver with url "https://foo.bar"
```

Reply:
ahhh i see now, thanks!